### DISCUSSION TITLE
Module parse failed: Unexpected token

### DISCUSSION THREAD

**Discussion by LessThanThreeBikes** [quasarframework/quasar#14747] [General - Components / Directives / etc]
I somehow messed up my Quasar v1 environment.  I am getting the "Module parse failed: Unexpected token" error. It looks like webpack is throwing an error on lines where I use optional chaining.  I tried to rollback my repo back to a known good check-in and reinstalled node_module but the problem persists.  Any guidance is greatly appreciated.

**Comment by suleimanas** [comment]


**Reply by LessThanThreeBikes** [reply]
It is not clear that this is relevant to my situation.  Some of the config files referenced in you link do not exist with a Quasar install.  Optional chaining was working fine until I installed the capacitor/push-notification plugin.  Installing this plugin seems to clobber something that is cached because I am not seeing any config files changing aside from package.json which shows the module. And note, rolling back to a know good commit doesn't resolve the issue.

**Reply by LessThanThreeBikes** [reply]
Is it the same as what?  I didn't upgrade my version of Node so it is the same as before I started experiencing this problem.
I am running NodeJS v16.15.0.  I see nowhere in the Webpack 4 documentation that it is only compatible or even recommended to use NodeJS v12.

Regardless, downgrading to v12 is not reasonable.  Everything was working on 16.  I added a module (@capacitor/push-notifications) and then something behinds the scenes changed. I say "behind the scenes" because git identified the package being added as the only change to my project.  I should be able to roll back, but the problem persists. I am trying to figure this part out.  

Incidentally, I can clone my project to a new folder, roll-back to an earlier commit and get things working again.  I just cannot get things working in my original project folder even after rolling back.  There seems to be something caching behind the scenes.  I would love to know what is happening to understand things better. I was hoping someone else recognized this issue and could illuminate me. Falling that, I will have to set my project aside to scaffold something up to catch what is going on.

I appreciate you help anyway.
